src/v/raft/offset_translator.cc

```diff
@@ -104,7 +104,7 @@ offset_translator::start(must_reset reset, bootstrap_state&& bootstrap) {
     if (reset) {
         vlog(_logger.info, "resetting offset translation state");
 
-        _state = storage::offset_translator_state(
+        *_state = storage::offset_translator_state(
```

Comment:
> When `raft::offset_translator` updated state during
`prefix_truncate_reset` all the users which cached the `_state` pointer

nice catch. why would they have `operator=(T&&)` defined for the shared pointer? that seems really odd.

would the offset translator api be easier to use if any wanting to hold a reference to the state held a reference to the translator instead?

Reply:
yeah the `operator(T&&)` is indeed confusing. I was thinking about changing the interface to return a reference but i do not know all the lifecycle implications. There may be a user that is relaying on the shared pointer semantics.

Reply:
@mmaslankaprv 

> There may be a user that is relaying on the shared pointer semantics.

Isn't this untrue now that you removed all cases of users keeping a copy of the shared pointer?